rest-services/it.tests/src/test/java/com/_4point/aem/docservices/rest_services/it_tests/server/forms/ExportDataTest.java

```diff
@@ -0,0 +1,81 @@
+
+package com._4point.aem.docservices.rest_services.it_tests.server.forms;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.MatcherAssert.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static com._4point.aem.docservices.rest_services.it_tests.TestUtils.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.io.IOUtils;
+import org.glassfish.jersey.client.authentication.HttpAuthenticationFeature;
+import org.glassfish.jersey.media.multipart.FormDataMultiPart;
+import org.glassfish.jersey.media.multipart.MultiPartFeature;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com._4point.aem.docservices.rest_services.it_tests.ByteArrayString;
+import com._4point.aem.docservices.rest_services.it_tests.TestUtils;
+import com._4point.aem.fluentforms.api.Document;
+import com._4point.aem.fluentforms.impl.SimpleDocumentFactoryImpl;
+
+public class ExportDataTest{
+
+
+	private static final String EXPORT_DATA_URL = "http://" + TEST_MACHINE_NAME + ":" + TEST_MACHINE_PORT_STR + "/services/FormsService/ExportData";
+	private static final MediaType APPLICATION_PDF = new MediaType("application", "pdf");
+	
+	private static final boolean SAVE_RESULTS = false;
+	
+	private WebTarget target;
+
+	@BeforeEach
+	void setUp() throws Exception {
+		HttpAuthenticationFeature feature = HttpAuthenticationFeature.basic(TEST_USER, TEST_USER_PASSWORD);	// default AEM passwords
+		target = ClientBuilder.newClient()
+							  .register(feature)
+							  .register(MultiPartFeature.class)
+							  .target(EXPORT_DATA_URL);
+	}
+	 
+	@Test
+	void testExportData_Bytes() throws IOException {
+		
+	
+
+			
+
+		try (@SuppressWarnings("resource")
+		final FormDataMultiPart multipart = new FormDataMultiPart()
+													.field("pdforxdp", TestUtils.SAMPLE_FORM_XDP.toFile() ,APPLICATION_PDF)
+													.field("dataformat", MediaType.APPLICATION_XML)) {
+			
+			Response result = target.request()
+									  .accept(MediaType.APPLICATION_XML_TYPE)
+									  .post(Entity.entity(multipart, multipart.getMediaType()));
+			
+			assertTrue(result.hasEntity(), "Expected the response to have an entity.");
+			assertEquals(Response.Status.OK.getStatusCode(), result.getStatus(), ()->"Expected response to be 'OK', entity='" + TestUtils.readEntityToString(result) + "'.");
+			byte[] resultBytes = IOUtils.toByteArray((InputStream)result.getEntity());
+			assertThat("Expected a xml to be returned.", ByteArrayString.toString(resultBytes, 8), containsString("%, X, M, L"));
```

Comment:
This test doesn't seem correct either.  Have you run these tests successfully?  I am not able to.

The result I am getting is:
```xml
<?xml version="1.0" encoding="UTF-8"?><form1>
    <TextField1/>
    <TextField2/>
</form1>
```

That result looks correct, but the result validation is wrong.  This looks like a slight modification of the PDF validation which don't really apply to this case (as we're expecting XML here).  I would instead just make sure that what we're getting back is valid XML (so try something like loading it into an XML DOM).

Reply:
Hey Rob,

I am not able to run any of the integration tests in my local. All the tests are failing.

Thanks,
Ritika.

Reply:
Hi Ritika,

Do you have access to a running AEM server with the fluentforms.core and the rest-services.server bundles installed on it?  That is a requirement for the integration tests to run.

By default, the code assumes AEM is running locally on port 4502.  This can be changed by altering the values of TEST_MACHINE_NAME and TEST_MACHINE_PORT in the TestUtils class.

If you have these two projects building locally using Maven, you can use Maven to install the bundles using the autoInstallBundle profile.  This uses different values (properties in the pom.xml file) to locate the AEM server, but the default is also localhost and port 4502.

Alternatively, once the jars for fluentforms.core and rest-services.server are built, you can also drop them into the crx-quickstart/install directory however I prefer to use maven.

Once the AEM server is available with the bundles on it, I believe all the integration tests should run successfully.  There is an outstanding issue where, when the AEM server is remote and is running on a different OS than the client machine, some tests fail.  This is because the paths being created locally have the wrong separator when interpreted by the server.

If you encounter issues where certain specific tests fail to run, let me know.  There may be some tests that rely on a specific sample form being loaded on the server, but as much as possible, I try and avoid that in the tests.  If all the tests are failing, then it's something more fundamental at fault.

Let me know if you can't get the tests running and I will try and help.

Regards,
Rob